http/user_resource_mapping_service.go

```diff
@@ -278,13 +278,15 @@ func (s *UserResourceMappingService) FindUserResourceMappings(ctx context.Contex
 		return nil, 0, err
 	}
 
-	urs := make([]*influxdb.UserResourceMapping, len(results.Users))
-	for k, item := range results.Users {
-		urs[k] = &influxdb.UserResourceMapping{
-			ResourceID:   f.ResourceID,
-			ResourceType: f.ResourceType,
-			UserID:       item.User.ID,
-			UserType:     item.Role,
+	urs := make([]*influxdb.UserResourceMapping, 0, len(results.Users))
+	for _, item := range results.Users {
+		if !f.UserID.Valid() || f.UserID != item.User.ID {
```

Comment:
where is the test for this?

Reply:
For the client?

There's no test for the http client I think

Reply:
I can create one as part of this PR 👍

Reply:
That brings up another question, why isn't the server doing the filtering?